Sources/SourceKitLSP/Workspace.swift

```diff
@@ -114,6 +114,7 @@ public final class Workspace {
           databasePath: dbPath.pathString,
           library: lib,
           delegate: indexDelegate,
+          useExplicitOutputUnits: true,
```

Comment:
if no units are ever set, does indexing still work?

Reply:
no, 0 visibility in that case
I think we need to figure out a way for user to specify whether they want opt in scheme based index visibility or not

Reply:
can we set this later?

Reply:
no, it's only exposed via initializer
dunno if it's viable through c++, but would require some indexdatabase PR

Reply:
Two quick approaches to this:
- Add something like `SchemeIndexVisibility` to build server json spec, then initialize indexdb accordingly
- Add something to client extension config

Reply:
If we can do this only for the build server build system that is fine, some kind of interface on the build system to register whether this is supported or not.

> another problem is when scheme is empty, without changes in indexdb we need additional call to fetch all targets and import everything, this comes with a huge payload and one round loop

Would prefer to have no index in this case I think, and just hint users that they have to select a scheme for indexing to work. If we do want to support this, the only thing that makes sense is to make the property adjustable outside the initializer, which is probably the easiest approach to getting this working anyway.

Reply:
I see internally its being used as a switch, so you have 2 different possibilities:

- use explicit outputs and output paths
- use source paths and an empty list means make everything visible

Seems like first option is the new world, latter is what is being used right now. I think its fine to use this as an incentive to get developers to choose a scheme, and potentially we just autoselect the scheme for the first file they open.

Reply:
> Would prefer to have no index in this case I think,
Problem is ide doesn't necessarily have the ability to choose scheme, an option would probably be nice

I'm gonna add an option to the language server, initialize the indexdb  with/without explicit mode accordingly, the rest of the implementation don't need to change.

Reply:
Good enough for now, the rest can come out when upstreaming